integration/utils/follower.go

```diff
@@ -139,16 +135,21 @@ Loop:
 			}
 			for _, tx := range col.TransactionIDs {
 				blockTxs++
-				if ch, loaded := f.txToChan.LoadAndDelete(tx.Hex()); loaded {
-					txi := ch.(txInfo)
 
+				f.mu.Lock()
```

Comment:
could you refactor the this into a method using the lock / defer unlock pattern

it's easy for future dev to shoot themselves in the foot with inline locking.